mplc/doc/documentation.md

```diff
@@ -0,0 +1,497 @@
+                       methods=['Shapley values'])
+```
+
+Once ran, accessing the results is straightforward:
+
+```python
+contributivity_score = my_scenario.contributivity_list
+print(contributivity_score[0])
+```
+
+Check out [Tutorial 4 - Exploring contributivity](https://github.com/SubstraFoundation/distributed-learning-contributivity/blob/master/notebooks/examples/4_Exploring_contributivity.ipynb) for more information.
+
+> Note: there is a lot more parameters to play with, which are fully explained below.
+
+## Scenario parameters
+
+In this documentation, available scenario parameters are structured into the following subsections:
+
+- [Choice of dataset](#choice-of-dataset)
+- [Definition of collaborative scenarios](#definition-of-collaborative-scenarios)
+- [Configuration of the collaborative and distributed learning](#configuration-of-the-collaborative-and-distributed-learning)
+- [Configuration of contributivity measurement methods to be tested](#configuration-of-contributivity-measurement-methods-to-be-tested)
+- [Miscellaneous](#miscellaneous)
+
+### Choice of dataset
+
+There are 2 ways to select a dataset. You can either choose a pre-implemented dataset, by setting the `dataset_name` parameter, or create a custom dataset object and pass it to the `dataset` parameter. To look at the structure of the dataset object, see the [Dataset generation](#dataset-generation) section below.
+
+- `dataset`: `None` (default), `datasets.Dataset object`  
+  This enables to use your own dataset, with the class Dataset. The [Tutorial 2](https://github.com/SubstraFoundation/distributed-learning-contributivity/blob/master/notebooks/examples/2%20_Sentiment140.ipynb) provides more explicit information.
+  If None, the dataset provided by the `dataset_name` parameter will be used.  
+  Note about validation and test datasets:
+
+    - The dataset modules must provide separated train and test sets (referred to as global train set and global test set).
+    - The global train set is then further split into a global train set and a global validation set.
+    - In the multi-partner learning computations, the global validation set is used for early stopping and the global test set is used for performance evaluation.
+    - The global train set is split amongst partner (according to the scenario configuration) to populate the partner's local datasets.
+    - For each partner, the local dataset can be split into separated train, validation and test sets, depending on the dataset configuration. Currently, the local validation and test set are not used, but they are available for further developments of multi-partner learning and contributivity measurement approaches.
+
+- `dataset_name`: `'mnist'` (default), `'cifar10'`, `'esc50'`, `'imdb'` or `'titanic'`  
+  MNIST, CIFAR10, ESC50, IMDB and Titanic are currently supported. They come with their associated modules in `/mplc/datasets` for loading data, pre-processing inputs, and define a model architecture.
```

Comment:
Now these datasets are subclass of the Dataset object, and are in the same dataset.py file

Reply:
Well spotted